Store onboarding: Update payment task description, priority and URL

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10570 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR updates a few things for the payment onboarding task:
- Prioritize it over the custom domain and launch store tasks.
- New description following 7abnJ5szUDhwiczNwhcyW3-fi
- Updated the URL to open for `woocommerce-payment` task following pe5sF9-1VU-p2.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
- Log in to the app.
- Open a free trial site or create one if needed.
- Navigate to the My Store screen, notice the payment task on the onboarding task list is above the custom domain and launch store tasks. The description should be correct.
- Tap the task > Continue Setup. A web view should be displayed showing the WooPayments page with the Finish Setup button displayed.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/e3a964c0-d878-4d0f-a3c6-83b98ab06187" width=320 /> <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/b2d01171-6809-4b46-8f47-a6a01f075c50" width=320 />

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Update link to open for wcpay onboarding task

itsmeichigo committed:
Update priority for payment onboarding task

itsmeichigo committed:
Update description for payment onboarding task

itsmeichigo committed:
Update release notes

selanthiraiyan commented on Networking/Networking/Model/StoreOnboardingTask.swift:
```diff
@@ -60,11 +60,11 @@ private extension StoreOnboardingTask.TaskType {
         case .addFirstProduct:
             return 2
-        case .launchStore:
+        case .payments, .woocommercePayments:
```
Nit: Kindly consider adding a unit test to check the sort order.

itsmeichigo replied:
These are already tested in `StoreOnboardingViewModelTests` and failed - updated in 2d56e7b2e506f4928895c5f63cc684b7128b4197.

selanthiraiyan reviewed (approved):
This works as expected. 🚀

itsmeichigo committed:
Update failed unit tests

itsmeichigo committed:
Fix test failure in StoreOnboardingTasksStoreTests

itsmeichigo committed:
Update URL for the wcpay task

itsmeichigo committed:
Different wording for payment and woopay tasks

itsmeichigo commented:
I added a few more changes:
- Changed the URL for the `woocommerce-payments` task following pe5sF9-1Xb-p2#comment-2813.
- Restored the description for the `payment` onboarding task.

Please feel free to test again @selanthiraiyan, I'll merge this PR tomorrow if there's no concern about these.

selanthiraiyan commented:
The new commits look good to me. 🚀

itsmeichigo committed:
Merge branch 'trunk' into feat/10570-payment-onboarding-task-update

itsmeichigo committed:
Merge branch 'trunk' into feat/10570-payment-onboarding-task-update
